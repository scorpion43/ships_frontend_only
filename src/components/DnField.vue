<template>
  <div class="dn-field"
    :class="{'just-clicked': justClicked}"
    @click="click">
    <!-- {{ cordinates }} -->
  </div>
</template>
<script>
export default {
  name: "DnField",
  props: {
    cordinates: {
      type: String,
      required: true,
    }
  },
  data() {
    return {
      justClicked: false,
    }
  },
  methods: {
    click () {
      this.justClicked = true
      this.$emit('click', this.cordinates)
      setTimeout(() => {
        this.justClicked = false
      }, 800)
    }
  }
}
</script>
<style scoped lang="scss">
  .dn-field {
    width: 50px;
    height: 50px;
    border: 1px solid black;
    &.just-clicked {
      background: yellow;
    }
  }
</style>