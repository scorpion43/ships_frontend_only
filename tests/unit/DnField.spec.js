import DnField from "@/components/DnField"
import { mount } from "@vue/test-utils"
import sinon from 'sinon' 

describe('DnField', () => {
  it('should emit click with cordinate A1', async () => {
    const wrapper = mount(DnField, {
      propsData: {
        cordinates: 'A1'
      }
    })

    await wrapper.trigger('click')
    await wrapper.vm.$nextTick()

    expect(wrapper.emitted().click).toBeTruthy()
    expect(wrapper.emitted().click.length).toBe(1)
    expect(wrapper.emitted().click[0][0]).toBe('A1')
  })

  it('should has backgrond white before click, after click yellow and after 800ms white back', async () => {
    const wrapper = mount(DnField, {
      propsData: {
        cordinates: 'A1'
      }
    })

    const clock = sinon.useFakeTimers() 

    expect(wrapper.element).toMatchSnapshot()

    await wrapper.trigger('click')
    await wrapper.vm.$nextTick()

    expect(wrapper.element).toMatchSnapshot()

    clock.tick(800)
    await wrapper.vm.$nextTick()

    expect(wrapper.element).toMatchSnapshot()
  })
})