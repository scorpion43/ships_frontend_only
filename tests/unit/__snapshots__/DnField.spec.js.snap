// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnField should has backgrond white before click, after click yellow and after 800ms white back 1`] = `
<div
  class="dn-field"
/>
`;

exports[`DnField should has backgrond white before click, after click yellow and after 800ms white back 2`] = `
<div
  class="dn-field just-clicked"
/>
`;

exports[`DnField should has backgrond white before click, after click yellow and after 800ms white back 3`] = `
<div
  class="dn-field"
/>
`;
